### ISSUE TITLE
Installation guide for Debian 10

### ISSUE THREAD

**Issue by karibuTW** [Flood-UI/flood#869]
Hello guys,
I revisited my 2y old tutorial as I had several requests to do so.

I am just sharing how I installed rtorrent and flood on a fresh debian 10
https://freedif.org/how-to-install-flood-rtorrent-on-debian-10

It works well like this, but I guess I can do further tweak.
If you have some amelioration ideas, please share.

Thank you

**Comment by Commissar0617** [comment]
-bash: npm: command not found

**Comment by karibuTW** [comment]
Hi, you are supposed to install it at step 4:

apt install curl build-essential git
curl -sL https://deb.nodesource.com/setup_12.x | sudo -E bash -
apt install -y nodejs

Any issue there?

Thanks

**Comment by Commissar0617** [comment]
honestly, im not sure if that did much, as i was doing this on an LXC container on root, hence no sudo. 
i just went and apt installed npm and it seems to have worked.

**Comment by Commissar0617** [comment]
so, i followed the instructions... but now all im getting is the default apache page...
both flood and rtorrent show as running.

**Comment by Commissar0617** [comment]
also ExecStop=/usr/bin/killall 

gives an error if it's in there, as there is no /usr/bin/killall

**Comment by karibuTW** [comment]
You may have a very minimal system
apt install psmisc

If this works, I will update the tutorial. Thanks

**Comment by Commissar0617** [comment]
still no flood UI on 80 or 3000.

**Comment by Commissar0617** [comment]
possible some progress... I added a proxypass to /node and it give a 503 now instead of 404

**Comment by Commissar0617** [comment]
https://github.com/Flood-UI/flood/wiki/Troubleshooting

the steps for being on a non-local got me in, but it won't connect to rtorrent now.

**Comment by karibuTW** [comment]
So you have access to flood UI now? What have you done?
And for the rtorrent, you use the right port?

**Comment by Commissar0617** [comment]
if i start flood manually, yes. not if it is started as a service.(which is weird because the flood service literally uses the same script, i verified that)

and i am using port 5000 for rtorrent, which is the port i set in the UI the one time i could access it.

**Comment by Commissar0617** [comment]
fixed the issue with it not running as a service, had file permission errors. i can now access the ui at port 3000, but it still can't talk to rtorrent

**Comment by karibuTW** [comment]
You have this in your .rtorrent.rc?
# SCGI port, used to communicate with Flood
network.scgi.open_port = 127.0.0.1:5000

It will open the communication port on 5000 for Flood to connect to it.

**Comment by Commissar0617** [comment]
yep. still nothing listening on 500,a nd netstat -tulpn shows no rtorrent

**Comment by Commissar0617** [comment]
> root@ruskidown:/home/rtorrent# service rtorrent status 
* rtorrent.service - rTorrent
   Loaded: loaded (/etc/systemd/system/rtorrent.service; enabled; vendor preset: enabled)
   Active: inactive (dead) since Mon 2020-03-30 01:41:13 UTC; 5s ago
  Process: 1209 ExecStart=/usr/bin/screen -d -m -fa -S rtorrent /usr/bin/rtorrent (code=exited, status=0/SUCCESS)
 Main PID: 1209 (code=exited, status=0/SUCCESS)
    Tasks: 0 (limit: 4915)
   Memory: 4.6M
   CGroup: /system.slice/rtorrent.service

Mar 30 01:41:13 ruskidown systemd[1]: Started rTorrent.
Mar 30 01:41:13 ruskidown systemd[1]: rtorrent.service: Succeeded.
root@ruskidown:/home/rtorrent# 

this look right?

**Comment by Commissar0617** [comment]
so im actually starting over using a VM instead of a ct. has issues with LXC privleges that wld keep it from working

**Comment by Jogai** [comment]
I'd recommend using nvm instead of nodejs from the debian repositories. Its friendlier for people that are new to node, and might have to change the node version for other projects.

**Comment by karibuTW** [comment]
Thanks Jogai for the comment.

**Comment by Jogai** [comment]
Well I saw later you tie into the services, I dont know how well an nvm managed node works with that. In my setup I start it manually.

**Comment by Luth1ng** [comment]
hey @karibuTW 
I followed your guide to make a fresh install of flood on my Debian10.

I also added the https functionnality with the apache virtualhost.
I can send you my config file if you want to add it to your guide

Thanks for your work!

**Comment by karibuTW** [comment]
Super, you used letsencrypt or different? I should cover it I guess, so I'll be happy to get info from you.

Thank you

**Comment by acantepie** [comment]
Thank for your work @karibuTW :+1: 
I have used your guide to install my seedbox on raspbian 10.

I my case :
 - i change floodServerHost to '0.0.0.0' on flood/config.js
 - i have to rebuild node.sass package cause of ARM arch (before execute npm run build)
 - i have to execute all npm commands with --unsafe-perm options
